apps/rush-lib/src/logic/policy/GitEmailPolicy.ts

```diff
@@ -0,0 +1,120 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+// See LICENSE in the project root for license information.
+
+import * as colors from 'colors';
+import * as os from 'os';
+
+import { GitPolicy } from './GitPolicy';
+import { RushConfiguration } from '../../api/RushConfiguration';
+import { AlreadyReportedError } from '../../utilities/AlreadyReportedError';
+import { Utilities } from '../../utilities/Utilities';
+
+interface IResultOrError<TResult> {
+  error?: Error;
+  result?: TResult;
+}
+
+export class GitEmailPolicy extends GitPolicy {
+  public static getUserEmail(rushConfiguration: RushConfiguration): string {
```

Comment:
>getUserEmail [](start = 16, length = 12)

Add some docs for this function to clarify its spec.  I think it should be like this:

getUserEmail() is called by someone who needs the email to exist.  If it is blank, or Git fails, or for any reason we can't get a usable string, then it should show a pretty message and throw AlreadyReportedError().

(If someone later needs to test whether the email is defined, then maybe we would expose tryGetEmail)